app/models/search_suggestion.rb

```diff
@@ -17,8 +17,12 @@ def cached_suggestions_for(term)
     end
 
     def start_with(term)
-      cached_suggestions_for(term).select { |s| s.value =~ /^#{term}/i }
-                                  .first(TradeTariffFrontend.suggestions_count)
+      cached_suggestions = cached_suggestions_for(term)
+
+      return [] unless cached_suggestions
```

Comment:
This isn't needed AFAIK

If the cached_suggestions_for method returns an empty Array when there are none (it definitely should do), then the select will iterate over nothing and return an empty Array and the first method will return an empty array, too.

Reply:
The issue is that  cached_suggestions can return nil,
in that case we get NoMethodError,
for example when you try to search "+" (the example in the screenshot above)